### DISCUSSION TITLE
Failing by telling docker-compose not installed

### DISCUSSION THREAD

**Discussion by Nagarjunakuchi88** [opencrvs/opencrvs-core#6351] [Technical Q&A] [answered]
Hi Team,
docker compose is available but failing by telling not available. Can you please guide us here
![Screenshot from 2023-12-08 09-13-36](https://github.com/opencrvs/opencrvs-core/assets/64436505/4dd7aee5-2303-4f59-814e-50f1b7a3e353)

**Comment by Zangetsu101** [comment, marked as answer]
Hi @Nagarjunakuchi88. It looks like the script was unable to find "docker-compose". We are aware that "docker-compose" is now deprecated and we will also be moving to "docker compose" starting with v1.4.0 at the latest.

**Reply by Nagarjunakuchi88** [reply]
Thank you for your quick answer. I am able to go ahead now.

Now while getting error while user creation or updation.
![Screenshot from 2023-12-11 06-58-49](https://github.com/opencrvs/opencrvs-core/assets/64436505/b52d109c-0943-4896-a716-fdddb67aa7ed)

**Reply by Zangetsu101** [reply]
Could you tell us which version of OpenCRVS are you trying to run? For both the `opencrvs-core` & `opencrvs-farajaland` repositories please.

In the meantime, please follow these steps & see if it solves your issue:
1. start the opencrvs-core & farajaland repo separately, you can do so by running `yarn dev` inside the root directory of the repos.
2. once all the services are up and running, you can check so using https://is-my-opencrvs-up.netlify.app/, in a separate terminal try running the following commands:
```bash
cd /path/to/opencrvs-core
yarn db:clear:all && yarn seed:dev
```

**Reply by NagarjunaKuchi** [reply]
Hi @Zangetsu101 I am following the steps mentioned here.

https://documentation.opencrvs.org/setup/3.-installation/3.1-set-up-a-development-environment

Deploying from the master branch.

**Reply by Zangetsu101** [reply]
okay that's great. In that case, see if you can start OpenCRVS core & country-config locally by following these instructions: https://documentation.opencrvs.org/setup/3.-installation/3.1-set-up-a-development-environment/3.1.3-starting-and-stopping-opencrvs.

Then check if all the services are running by going here: https://is-my-opencrvs-up.netlify.app/

If all of them are running ok, then try the commands I mentioned above. Let us know if the problem still persists

**Comment by NagarjunaKuchi** [comment]
@Zangetsu101 I followed above steps. Now it's hanging at below step.
![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/d379f0b5-0613-40ef-96f3-576e95bf89e7)

![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/fd30a3a0-0db0-472d-9367-50c3d06cf2b9)

**Reply by Zangetsu101** [reply]
This is what the "opencrvs-core" would show once it has started. What do you see once you open:  https://is-my-opencrvs-up.netlify.app/ at this point?

**Reply by Nagarjunakuchi88** [reply]
![Screenshot from 2023-12-12 14-59-59](https://github.com/opencrvs/opencrvs-core/assets/64436505/7a9cadd6-6619-40d5-99b7-348574df5b26)
opencrvs-core:

![Screenshot from 2023-12-12 15-00-40](https://github.com/opencrvs/opencrvs-core/assets/64436505/fe72c3ee-34f0-4318-a348-60d26f5e2ceb)
opencrvs-config:

Configuration is failing at intial stage . When configuration stats opencrvs-core is failing with connection refused error.

[1] openhim-core_1     | 2023-12-12T09:38:30.456Z [worker1] error: Error: connect ECONNREFUSED 172.17.0.1:5050
[1] openhim-core_1     |     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1148:16)


![Screenshot from 2023-12-12 15-01-34](https://github.com/opencrvs/opencrvs-core/assets/64436505/3174cb64-c9d9-415c-9f4b-c19d7ee028d9)

**Comment by Zangetsu101** [comment]
running `yarn` inside the "opencrvs-farajaland" directory should fix the issue with "cross-env". 

But I wonder what's causing openhim to not start up properly

**Reply by Nagarjunakuchi88** [reply]
I think we are running yarn inside "opencrvs-farajaland" .

![Screenshot from 2023-12-12 15-28-54](https://github.com/opencrvs/opencrvs-core/assets/64436505/cd729de1-07a6-40f5-98ec-58c13aefb15f)

**Comment by Zangetsu101** [comment]
I meant the `yarn install` command, a shorthand for which is just `yarn`. It installs all the dependencies for a project

**Reply by Nagarjunakuchi88** [reply]
Ya got it

**Reply by Nagarjunakuchi88** [reply]
Now configuration is up. But failing for login with Invalid username and password.

AppStatus:
![Screenshot from 2023-12-12 15-47-31](https://github.com/opencrvs/opencrvs-core/assets/64436505/c8dbccd8-94ce-44e3-98ea-ea49f1401980)

Login Error:
![Screenshot from 2023-12-12 15-47-58](https://github.com/opencrvs/opencrvs-core/assets/64436505/8d2228f4-7152-4d9a-a416-c9f904422c99)

opencrvs-core console :
![Screenshot from 2023-12-12 15-48-31](https://github.com/opencrvs/opencrvs-core/assets/64436505/9387e054-c07b-403f-858e-4113c75881be)

**Reply by Nagarjunakuchi88** [reply]
OpenHIM console credentials seems not working.

![Screenshot from 2023-12-12 16-42-08](https://github.com/opencrvs/opencrvs-core/assets/64436505/e946b1d9-a5f9-48e7-8ec3-49c03288722d)

**Reply by Zangetsu101** [reply]
okay that is expected because we haven't run `data-seeding` yet. Now inside "opencrvs-core" directory run this command:
```yarn db:clear:all && yarn seed:dev```

You should be able to login with the default users after the seeding finished successfully

**Reply by Nagarjunakuchi88** [reply]
With the above step, we are getting the same issue when we ran bash setup.sh.

![Screenshot from 2023-12-12 17-32-55](https://github.com/opencrvs/opencrvs-core/assets/64436505/246dd4f1-3ee2-43c5-bfda-5af28982c59e)

**Reply by Zangetsu101** [reply]
could you try running just `yarn seed:dev` this time and see if it gives you the same error?

**Reply by Nagarjunakuchi88** [reply]
Yes it is giving the same error.

![Screenshot from 2023-12-12 17-41-20](https://github.com/opencrvs/opencrvs-core/assets/64436505/7d35a364-085d-4958-93de-918845d9b17a)

`{"level":50,"time":1702383431672,"pid":191446,"hostname":"techno251-Latitude-3490","req":{"id":"1702383431388:techno251-Latitude-3490:191446:lq27dznn:10206","method":"post","url":"/createUser","headers":{"content-type":"application/json","authorization":"Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzY29wZSI6WyJuYXRsc3lzYWRtaW4iLCJzeXNhZG1pbiJdLCJpYXQiOjE3MDIzODM0MzEsImV4cCI6MTcwMjk4ODIzMSwiYXVkIjpbIm9wZW5jcnZzOmF1dGgtdXNlciIsIm9wZW5jcnZzOnVzZXItbWdudC11c2VyIiwib3BlbmNydnM6aGVhcnRoLXVzZXIiLCJvcGVuY3J2czpnYXRld2F5LXVzZXIiLCJvcGVuY3J2czpub3RpZmljYXRpb24tdXNlciIsIm9wZW5jcnZzOndvcmtmbG93LXVzZXIiLCJvcGVuY3J2czpzZWFyY2gtdXNlciIsIm9wZW5jcnZzOm1ldHJpY3MtdXNlciIsIm9wZW5jcnZzOmNvdW50cnljb25maWctdXNlciIsIm9wZW5jcnZzOndlYmhvb2tzLXVzZXIiLCJvcGVuY3J2czpjb25maWctdXNlciIsIm9wZW5jcnZzOmRvY3VtZW50cy11c2VyIl0sImlzcyI6Im9wZW5jcnZzOmF1dGgtc2VydmljZSIsInN1YiI6IjY1Nzg0YjQ5MmJjOGU5MjdiMzFjYTk4MyJ9.mrmGwYjdvZwoinLxaBx4jJyXBq8cK8SCVWybtw05kou_48nFoU3TYDKEFv5ztm4RcqYcb9UwtXseeH_37eNvLgjCHCUzUZs0qK1UMAC03SKTGI0JDWmIgriMeX3jvAsh0WravNwUae0YKSDfcQ_LoqzHwqyzo7dMDCFIMFs1R3yFqy391OtDAyElr_vb0R62ZqYyqwj1DKF6TzLBoAsjdAuRFc2YxoZzWimVESkBu4pv7P4y1f0wpQKjbdsteM6OfqLp4C6MD3sUUF_kDl5fx4oQzws23O4XrH0JKuK23doqk7N3TYSDOydHOqplvnPnaE11he0heeb5nw2VrUJzGw","x-correlation-id":"76","x-real-ip":"127.0.0.1","x-real-user-agent":"node-fetch/1.0 (+https://github.com/bitinn/node-fetch)","accept":"*/*","content-length":"343","user-agent":"node-fetch/1.0 (+https://github.com/bitinn/node-fetch)","accept-encoding":"gzip,deflate","host":"localhost:3030","connection":"keep-alive"},"remoteAddress":"127.0.0.1","remotePort":49754},"tags":["internal","error"],"err":{"type":"Error","message":"Unexpected response received","stack":"Error: Unexpected response received\n    at /home/techno-251/opencrvs-core/packages/user-mgnt/src/features/createUser/service.ts:167:11\n    at Generator.next (<anonymous>)\n    at fulfilled (/home/techno-251/opencrvs-core/packages/user-mgnt/src/features/createUser/service.ts:5:58)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)","isBoom":true,"isServer":true,"data":null,"output":{"statusCode":500,"payload":{"statusCode":500,"error":"Internal Server Error","message":"An internal server error occurred"},"headers":{}}},"msg":"Unexpected response received"}`

**Reply by Zangetsu101** [reply]
Let's check if all the migrations ran properly or not. In opencrvs-core,
```
cd packages/migration
yarn start
```
It should show you some tables with 2 columns "filename" & "applied at". Could you check if all the migration ran successfully, if one of them failed it would have have "pending" value at the "applied at" column.

Sorry for all the back & forth, it's that I'm trying to figure out what might be the cause of this issue

**Reply by Nagarjunakuchi88** [reply]
No issues. You are helping me out. Thank you for that.

It seems all migrations are completed.
<pre><font color="#4E9A06"><b>techno-251@techno251-Latitude-3490</b></font>:<font color="#3465A4"><b>~/opencrvs-core/packages/migration</b></font>$ yarn start
<b>yarn run v1.22.21</b>
<font color="#1E2224">$ yarn build &amp;&amp; ./runMigrations.sh</font>
<font color="#1E2224">$ rimraf build &amp;&amp; tsc</font>
./runMigrations.sh: 17: Bad substitution
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo up --file ./build/dist/src/migrate-mongo-config-hearth.js</font>
(node:208391) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo status --file ./build/dist/src/migrate-mongo-config-hearth.js</font>
(node:208425) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#555753">┌──────────────────────────────────────────────────────────────────┬──────────────────────────┐</font>
<font color="#555753">│</font><font color="#CC0000"> Filename                                                         </font><font color="#555753">│</font><font color="#CC0000"> Applied At               </font><font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20220817082035-informant-legal-guardin-doc.js                    <font color="#555753">│</font> 2023-12-12T12:00:03.801Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20220822085901-rejected-declarations-started-by.js               <font color="#555753">│</font> 2023-12-12T12:00:03.854Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221017133354-restore-supporting-document.js                    <font color="#555753">│</font> 2023-12-12T12:00:03.860Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221031131452-add_practioner_id_event_registrations.js          <font color="#555753">│</font> 2023-12-12T12:00:03.875Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221110092434-add_identifiers_DOB_gender_location.js            <font color="#555753">│</font> 2023-12-12T12:00:03.958Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221115062056-update-status-in-complete-fields.js               <font color="#555753">│</font> 2023-12-12T12:00:03.975Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221128065238-migrate-documents-to-minio.js                     <font color="#555753">│</font> 2023-12-12T12:00:04.059Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230127063226-update-practitioner-role.js                       <font color="#555753">│</font> 2023-12-12T12:00:04.346Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230220081813-update-indices.js                                 <font color="#555753">│</font> 2023-12-12T12:00:04.699Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230221105416-create-performance-views.js                       <font color="#555753">│</font> 2023-12-12T12:00:04.768Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230328065851-change-certified-status-to-issused.js             <font color="#555753">│</font> 2023-12-12T12:00:04.773Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230331182109-modify-minio-bucket-policy.js                     <font color="#555753">│</font> 2023-12-12T12:00:04.789Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230602090436-update-identifier-with-system.js                  <font color="#555753">│</font> 2023-12-12T12:00:04.898Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230602120944-populate-valid-locationid.js                      <font color="#555753">│</font> 2023-12-12T12:00:04.914Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230612060923-update-patient-identifier-with-codeableconcept.js <font color="#555753">│</font> 2023-12-12T12:00:04.929Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230905130632-elastic-search-property-rename.js                 <font color="#555753">│</font> 2023-12-12T12:00:04.937Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20231025175023-fix-crude-birth-rates.js                          <font color="#555753">│</font> 2023-12-12T12:00:04.944Z <font color="#555753">│</font>
<font color="#555753">└──────────────────────────────────────────────────────────────────┴──────────────────────────┘</font>
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo up --file ./build/dist/src/migrate-mongo-config-openhim.js</font>
(node:208456) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo status --file ./build/dist/src/migrate-mongo-config-openhim.js</font>
(node:208497) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#555753">┌─────────────────────────────────────────────────┬──────────────────────────┐</font>
<font color="#555753">│</font><font color="#CC0000"> Filename                                        </font><font color="#555753">│</font><font color="#CC0000"> Applied At               </font><font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20220825071508-channels-v1.0.1.js               <font color="#555753">│</font> 2023-12-12T12:00:06.205Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20220826062206-channels-v1.0.1-to-v1.1.0.js     <font color="#555753">│</font> 2023-12-12T12:00:06.235Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221011150329-channels-v1.1.0-to-v1.2.0.js     <font color="#555753">│</font> 2023-12-12T12:00:06.240Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221121124457-remove-old-transactions.js       <font color="#555753">│</font> 2023-12-12T12:00:06.246Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221124202628-user-audit-channels.js           <font color="#555753">│</font> 2023-12-12T12:00:06.262Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230131130400-add-certificate-issue-channel.js <font color="#555753">│</font> 2023-12-12T12:00:06.269Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230210112528-marriage-channels.js             <font color="#555753">│</font> 2023-12-12T12:00:06.294Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230328101901-potential-duplicate.js           <font color="#555753">│</font> 2023-12-12T12:00:06.301Z <font color="#555753">│</font>
<font color="#555753">└─────────────────────────────────────────────────┴──────────────────────────┘</font>
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo up --file ./build/dist/src/migrate-mongo-config-application-config.js</font>
(node:208528) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo status --file ./build/dist/src/migrate-mongo-config-application-config.js</font>
(node:208559) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#555753">┌──────────────────────────────────────────────────────────┬──────────────────────────┐</font>
<font color="#555753">│</font><font color="#CC0000"> Filename                                                 </font><font color="#555753">│</font><font color="#CC0000"> Applied At               </font><font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221205071415-add_health_facility_to_formdataset.js     <font color="#555753">│</font> 2023-12-12T12:00:07.481Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221212134153-update-systems.js                         <font color="#555753">│</font> 2023-12-12T12:00:07.509Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221221093004-store-informant-sms-notification.js       <font color="#555753">│</font> 2023-12-12T12:00:07.536Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230113115411-country-background.js                     <font color="#555753">│</font> 2023-12-12T12:00:07.541Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230208081755-add-marriage-configuration.js             <font color="#555753">│</font> 2023-12-12T12:00:07.546Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230301124048-add-marriage-certificate-configuration.js <font color="#555753">│</font> 2023-12-12T12:00:07.559Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230518055204-upload-certificate-templates-to-minio.js  <font color="#555753">│</font> 2023-12-12T12:00:07.648Z <font color="#555753">│</font>
<font color="#555753">├──────────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230825150509-remove-unnecessary-application-config.js  <font color="#555753">│</font> 2023-12-12T12:00:07.653Z <font color="#555753">│</font>
<font color="#555753">└──────────────────────────────────────────────────────────┴──────────────────────────┘</font>
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo up --file ./build/dist/src/migrate-mongo-config-user-mgnt.js</font>
(node:208590) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo status --file ./build/dist/src/migrate-mongo-config-user-mgnt.js</font>
(node:208621) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#555753">┌─────────────────────────────────────────────────────┬──────────────────────────┐</font>
<font color="#555753">│</font><font color="#CC0000"> Filename                                            </font><font color="#555753">│</font><font color="#CC0000"> Applied At               </font><font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221212134201-update-systems.js                    <font color="#555753">│</font> 2023-12-12T12:00:08.840Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20221229110512-modify-user-roleAndtype-fieldName.js <font color="#555753">│</font> 2023-12-12T12:00:08.868Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230315074827-create-integrating-system-type.js    <font color="#555753">│</font> 2023-12-12T12:00:08.873Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230511082648-drop-usernamerecords-collection.js   <font color="#555753">│</font> 2023-12-12T12:00:08.880Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230522081120-make-users-status-pending.js         <font color="#555753">│</font> 2023-12-12T12:00:08.885Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230615104846-create-opencrvs-super-user.js        <font color="#555753">│</font> 2023-12-12T12:00:09.044Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230623175740-migrate-profile-image-to-minio.js    <font color="#555753">│</font> 2023-12-12T12:00:09.139Z <font color="#555753">│</font>
<font color="#555753">├─────────────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20231106130956-email-mobile-unique-index.js         <font color="#555753">│</font> 2023-12-12T12:00:09.233Z <font color="#555753">│</font>
<font color="#555753">└─────────────────────────────────────────────────────┴──────────────────────────┘</font>
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo up --file ./build/dist/src/migrate-mongo-config-performance.js</font>
(node:208652) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#1E2224">$ /home/techno-251/opencrvs-core/node_modules/.bin/migrate-mongo status --file ./build/dist/src/migrate-mongo-config-performance.js</font>
(node:208683) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
<font color="#555753">┌────────────────────────────────────────────┬──────────────────────────┐</font>
<font color="#555753">│</font><font color="#CC0000"> Filename                                   </font><font color="#555753">│</font><font color="#CC0000"> Applied At               </font><font color="#555753">│</font>
<font color="#555753">├────────────────────────────────────────────┼──────────────────────────┤</font>
<font color="#555753">│</font> 20230221121625-create-performance-views.js <font color="#555753">│</font> 2023-12-12T12:00:10.385Z <font color="#555753">│</font>
<font color="#555753">└────────────────────────────────────────────┴──────────────────────────┘</font>
Done in 10.92s.
</pre>

**Reply by Zangetsu101** [reply]
I was trying to reproduce this issue but no avail. The seeding ran without any errors when I tried it on the master branch. I wonder if we are missing something

**Reply by Nagarjunakuchi88** [reply]
Okay.. let me retry again by cloning from the start.Seems only data seeder is the issue. Is there any way can i debug ?

**Reply by NagarjunaKuchi** [reply]
@Zangetsu101 tried all the ways but no use. Is there any chance can we skip this step and manually feed the data

**Reply by NagarjunaKuchi** [reply]
@Zangetsu101 I tried in another machine which is having below configuration. In this, it's reaching to last step and getting stuck there.

![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/37f4daed-aa2b-4f72-a00d-33c300cdd003)


![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/c50ce55d-08ab-4db0-b74b-2c1e0ad6b9ed)

![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/4620be1d-470b-4bef-bd5e-c1ee4953c526)

![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/257508c7-3d07-4dd6-97f3-c24b5f43d6e1)

When I ran on yarn dev on "opencrvs-farajaland" getting below error. Not able to connect 
![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/854c5438-5604-471b-a8d7-bc9a3a860951)

but same when I curl, able to get it 
![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/ca861453-1fa3-4dcc-9031-34d72cdad6b2)

**Reply by Zangetsu101** [reply]
Could you try closing these running instances & start the core & farajaland repo separately like how you did it previously?

**Reply by Nagarjunakuchi88** [reply]
When I ran separately I am getting the error saying " Failed to fetch public key from core. Make sure core is running and you able to access http://localhost:4040/. well-known".

But when I curl it, able to get public key.

**Reply by Zangetsu101** [reply]
By any chance, are you running this on windows or wsl?

**Reply by Nagarjunakuchi88** [reply]
No @Zangetsu101. These two machines are having Ubuntu only. One is server and another one is laptop, only this is the difference.

**Reply by Zangetsu101** [reply]
okay, I asked because I had faced a similar issue when running the project on wsl quite a while back. Anyway for some reason your farajaland instance is unable to connect to auth service running at port 4040

**Reply by Zangetsu101** [reply]
@Nagarjunakuchi88 Which version of node are you using?

**Reply by NagarjunaKuchi** [reply]
We are using 16.20.0

![image](https://github.com/opencrvs/opencrvs-core/assets/22051090/e9cc6e23-7b8f-4410-bafb-73517cec9397)

**Reply by Nagarjunakuchi88** [reply]
@Zangetsu101 able to deploy in server but still facing issues in the laptop.
![tmp_8d3ea639-85fc-4e82-b829-aae8f3b36588](https://github.com/opencrvs/opencrvs-core/assets/64436505/0ef909cc-1652-46a9-916b-ac2746367bb8)

Thank you for your help

**Reply by Zangetsu101** [reply]
That's good news. Though I think you are trying to run the development environment in your server which is probably not what you want. You should read through this to get a better understanding: https://documentation.opencrvs.org/setup/3.-installation/3.3-set-up-a-server-hosted-environment

Now for the issue you are facing trying to set up the development environment in your laptop, we are currently unable to pinpoint the exact cause of it but we can offer you more dedicated support if you email us at team@opencrvs.org and please answer the following questions:  

Please tell us your full name and how you wish to be referred to.
Please tell us the name of the organisation or company you work for.
Are you working with a government on an active country implementation? ... If so, which country’s implementation are you working on?
Which of these describes your project best?

- Testing OpenCRVS for business development, educational purposes or for personal interest
- A research project to evaluate, configure and test OpenCRVS as a possible solution for a government, an active brief or proposal. (If so, tell us about your project, brief, the client and timeline if you can.)
- Actively installing and configuring OpenCRVS Core for production use in a country.
- Customising OpenCRVS Core code in production in a country

**Reply by NagarjunaKuchi** [reply]
Sure Will send a mail with above requirement. 

But in server we deployed but not able to access from outside with ip address in the internal network. Any suggestion on this?

I whitelisted the ip address. Changed the HOSTNAME, login url and config url to 0.0.0.0 instead of localhost.

I think apollographql is stopping for CORS.

**Reply by Zangetsu101** [reply]
@Nagarjunakuchi88 As I mentioned earlier, the steps you followed for setting up OpenCRVS in your server were that of the "development environment" which is meant to used in a local environment. For that reason, it doesn't have the routing configurations required for it to be accessible via the internet. For a server hosted environment, we have an extensive documentation here: https://documentation.opencrvs.org/setup/3.-installation/3.3-set-up-a-server-hosted-environment about the steps required for setting it up. Hope that helps
